### PR TITLE
darkframe correction, hist

### PR DESCRIPTION
**Pull request by rroohhh**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -23,3 +23,5 @@ measure_bw.sh
 *.blob
 *.blob.sz3
 *.ipynb
+
+a.py
```
Comment (anuejn) [inline .gitignore:27]: 🗑️ ♻️

File: calibrate.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+input:
+  type: WebcamInput
+  device: {{ webcam-device }}
+
+decode:
+  type: DualFrameRawDecoder
+  input: <input
+
+calibrate:
+  input: <cache
```
Comment (anuejn) [inline calibrate.yml:10]: this does not work?
also why are there the caches?

File: src/nodes_gpu/calibrate.glsl [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    uint c = uint(source.data[raw_idx + 2]);
+
+    float first_value = float((a << 4) | (b >> 4)) + 128.0;
+    float second_value = float(((b << 8) & 0xf00) | c) + 128.0;
+
+    float corr_first_v = texelFetch(lut_sampler, pos * ivec2(2, 1), 0).r;
+    float corr_second_v = texelFetch(lut_sampler, pos * ivec2(2, 1) + ivec2(1, 0), 0).r;
+    uint corr_first = uint(round(first_value - corr_first_v));
+    uint corr_second = uint(round(second_value - corr_second_v));
+    // int corr_second = int(second_value);
```
Comment (anuejn) [inline src/nodes_gpu/calibrate.glsl:35]: 🗑️ ♻️

File: src/nodes_gpu/stats.rs [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+        let command_buffer = builder.build()?;
+
+        let future =
+            fut.then_execute(self.queue.clone(), command_buffer)?.then_signal_fence_and_flush()?;
+
+        future.wait(None).unwrap();
+        Ok(Payload::from(Frame {
+            interp: Raw {
+                width: 4096,
+                height: 1,
```
Comment (anuejn) [inline src/nodes_gpu/stats.rs:150]: I am a bit unsure if this is the right format to do this.
Also cant we name the node hist or something more specific?

File: src/nodes_gpu/calibrate.glsl [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    uint8_t a_corr = uint8_t(corr_first >> 4);
+    uint8_t b_corr = uint8_t(((corr_first << 4) & 0xf0) | (corr_second >> 8));
+    uint8_t c_corr = uint8_t(corr_second);
+
+    sink.data[raw_idx + 0] = a_corr;
+    sink.data[raw_idx + 1] = b_corr;
+    sink.data[raw_idx + 2] = c_corr;
+
+    /*
```
Comment (anuejn) [inline src/nodes_gpu/calibrate.glsl:53]: 🗑️ ♻️

**Review by anuejn — CHANGES_REQUESTED** [review]
Also: could you squash the two first commits?

File: .gitignore (hunk 1 above)
Comment (rroohhh) [reply .gitignore:27]: whats the problem with that?

File: src/nodes_gpu/stats.rs (hunk 4 above)
Comment (rroohhh) [reply src/nodes_gpu/stats.rs:150]: What format would you use?

File: .gitignore (hunk 1 above)
Comment (anuejn) [reply .gitignore:27]: Its just random? I thought mainly, that its a leftover and slipped in here by accident

File: src/nodes_gpu/stats.rs (hunk 4 above)
Comment (anuejn) [reply src/nodes_gpu/stats.rs:150]: I would use something more explicit. Like a Stats type

File: .gitignore (hunk 1 above)
Comment (rroohhh) [reply .gitignore:27]: well I mean its just a random script I used for testing, but I guess I can just put it into my local ignore settings

File: src/nodes_gpu/stats.rs (hunk 4 above)
Comment (rroohhh) [reply src/nodes_gpu/stats.rs:150]: Hmm yes that could work, but then you could no longer use the `RawBlobWriter` for example to write these to disk. 
Maybe this is a more general problem that needs a "proper" solution and not just this "hack".

**Review by anuejn — APPROVED** [review]

**Pull request merged**